src/libcore/benches/slice.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use test::black_box;
+use test::Bencher;
+
+#[bench]
+fn binary_search(b: &mut Bencher) {
+    let mut v = Vec::new();
+    for i in 0..999 {
+      v.push(i);
+    }
+    let mut i = 0;
+    b.iter(move || {
+        i += 1299827;
```

Comment:
Could we get a comment here as to what this number is intended to mean?

Reply:
It's a large (compared to 999) prime. Maybe I should use librand instead?

Reply:
This seems fine to me -- and better than random numbers when dealing with benchmarks -- but I would like a comment.

Reply:
I made it into a proper LCG and linked to where I got the constants.